src/GitVersion.Core/VersionCalculation/VersionSearchStrategies/TrunkBasedVersionStrategy.cs

```diff
@@ -114,77 +132,118 @@ public IEnumerable<BaseVersion> GetBaseVersions(EffectiveBranchConfiguration con
     {
         traversedCommits ??= [];
 
-        Lazy<IReadOnlyDictionary<ICommit, EffectiveBranchConfiguration>> commitsWasBranchedFromLazy = new(
-            () => GetCommitsWasBranchedFrom(branchName: iteration.BranchName)
-        );
+        bool exit = false;
```

Comment:
I'm not a fan of this `exit` variable. Could we solve this differently by removing it? The length of this method is also excessive, so a refactoring would do it good. Perhaps creating a (private) helper class would be good?

Reply:
Hmm good idea. I'm creating a ticket and do it in a separate PR if you not mind.